tools/build.py

```diff
@@ -230,6 +220,27 @@ def adjust_options(options):
     options.host_tuple = '%s-%s' % (platform.arch(), platform.os())
     options.target_tuple = '%s-%s' % (options.target_arch, options.target_os)
 
+    # Verify options consistency (sysroot vs. cross build)
+    if options.target_tuple != options.host_tuple:
+        # This is cross build, sysroot should/must be supplied
+        if options.target_os in ['nuttx', 'tizenrt']:
+            require_sysroot = True
+            options.buildlib = True
+        if not options.sysroot:
+            if require_sysroot:
+                ex.fail('--sysroot required for nuttx/tinyara')
```

Comment:
Is that tizenrt or tinyara? (The user is expected to specify the OS as tizenrt, so better give notifications in the same format.)

BTW, a second thought: even though right now only nuttx and tizenrt requires sysroot, should the build system evolve, others may require it, too. Then, this message will have to be updated again and again. Having an error string like `'--sysroot required for %s' % options.target_os` would be more future-proof (and would also solve the problem I've pointed out).